rewrite-core/src/main/java/org/openrewrite/trait/TypeReference.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.trait;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.SourceFile;
+import org.openrewrite.Tree;
+import org.openrewrite.TreeVisitor;
+
+import java.util.Set;
+
+public interface TypeReference extends Trait<Tree> {
+
+    String getName();
+
+    default boolean supportsRename() {
+        return false;
+    }
+
+    default TreeVisitor<Tree, ExecutionContext> renameTo(String name) {
+        throw new UnsupportedOperationException();
+    }
+
```

Comment:
I'd noticed these two are unused for now; would we still want those to be part of the initial spec?
If so I think we should implement these from SpringTypeReference already, as that's the use case here.

Reply:
We can also remove them for now. I added them as I wanted to validate the design. Implementing ChangeType felt like it would be too much work, so I stopped there.

Reply:
@Laurens-W I think we started down this route specifically to support change type right? Would it make sense to add this in before we merge? Otherwise we'd immediately have to change this again, across two repositories. Or would you want to not merge the SpringTypeReference here and have that in rewrite-spring only?

Reply:
I think merging this in the current state allows Knut to do some testing regarding the CLI while I can simultaneously make changes to `ChangeType`, I do agree that the Spring specific stuff should be moved to rewrite-spring before merging this though

Reply:
I think it would be nice if we can leave the SpringTypeReference in here until we can verify it works in CLI. I actually suspect it won't work, but I find it quite hard to test the CLI against unrelease OR changes. So I propose we merge this PR (with or without the methods above) and then I look into getting it working in CLI and SaaS. In a second step we can then move it. This also allows us to fix ChangeType without having to modify two repos.

Reply:
I would leave the Spring stuff in here for now. It doesn't really hurt and allows us to better validate the implementation before moving it. Also, it doesn't expose any public API, as SpringTypeReference is package private.